service/controller/resource/tccpn/create.go

```diff
@@ -0,0 +1,515 @@
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "requested the update of the tenant cluster's control plane nodes cloud formation stack")
+	}
+
+	return nil
+}
+
+func newAutoScalingGroup(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*template.ParamsMainAutoScalingGroup, error) {
+	autoScalingGroup := &template.ParamsMainAutoScalingGroup{
+		AvailabilityZone: key.ControlPlaneAvailabilityZones(cr)[0],
+		ClusterID:        key.ClusterID(&cr),
+		Subnet:           key.SanitizeCFResourceName(key.PrivateSubnetName(key.ControlPlaneAvailabilityZones(cr)[0])),
+	}
+
+	return autoScalingGroup, nil
+}
+
+func newEtcdVolume(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*template.ParamsMainEtcdVolume, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	etcdVolume := &template.ParamsMainEtcdVolume{
+		AvailabilityZone: key.ControlPlaneAvailabilityZones(cr)[0],
+		Name:             key.VolumeNameEtcdCP(cr),
+		SnapshotID:       cc.Status.TenantCluster.MasterInstance.EtcdVolumeSnapshotID,
+	}
+
+	return etcdVolume, nil
+}
+
+func newIAMPolicies(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane, encrypterBackend string) (*template.ParamsMainIAMPolicies, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var iamPolicies *template.ParamsMainIAMPolicies
+	{
+		iamPolicies = &template.ParamsMainIAMPolicies{
+			ClusterID:        key.ClusterID(&cr),
+			EC2ServiceDomain: key.EC2ServiceDomain(cc.Status.TenantCluster.AWS.Region),
+			RegionARN:        key.RegionARN(cc.Status.TenantCluster.AWS.Region),
+			S3Bucket:         key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID),
+		}
+
+		if encrypterBackend == encrypter.KMSBackend {
+			iamPolicies.KMSKeyARN = cc.Status.TenantCluster.Encryption.Key
+		}
+	}
+
+	return iamPolicies, nil
+}
+
+func newLaunchConfiguration(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*template.ParamsMainLaunchConfiguration, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	launchConfiguration := &template.ParamsMainLaunchConfiguration{
+		BlockDeviceMapping: template.ParamsMainLaunchConfigurationBlockDeviceMapping{
+			Docker: template.ParamsMainLaunchConfigurationBlockDeviceMappingDocker{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingDockerVolume{
+					Size: 100,
+				},
+			},
+			Kubelet: template.ParamsMainLaunchConfigurationBlockDeviceMappingKubelet{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingKubeletVolume{
+					Size: 100,
+				},
+			},
+			Logging: template.ParamsMainLaunchConfigurationBlockDeviceMappingLogging{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingLoggingVolume{
+					Size: 100,
+				},
+			},
+		},
```

Comment:
@calvix @paurosello we hard code this here. Is this legit? Does that have to be configurable? WDYT?

Reply:
All configuration is good but I would continue using current values for now

Reply:
I think current space is more than enough for everything

- Logging should be rotated or sent somewhere else if important
- Docker volume of 100Gb should be more than enough for master
- For the kubelet I don't know what is being stored in there

Reply:
hardcoding is fine as its  masters, I would just rather use some const so its easily changeable rather than literal numbers